testing/unittests/fxpptests.cpp

```diff
@@ -274,141 +210,80 @@ class fxppParserTests : public CppUnit::TestFixture
-            {
-              fprintf(stdout, "\nTest(%s) namespace attribute uri mismatch - '%s' instead of '%s'", cfg.name, stag.getURI(0), uri);
-              failed = true;
-            }
-            if (!streq(stag.getLocalName(1), "foo"))
-            {
-              fprintf(stdout, "\nTest(%s) unexpected local attribute name - '%s' instead of '%s'", cfg.name, stag.getLocalName(1), "foo");
-              failed = true;
-            }
-          }
+            CPPUNIT_ASSERT_EQUAL_MESSAGE("Local attribute name mismatch", std::string("xmlns"), std::string(stag.getLocalName(0)));
+
+          CPPUNIT_ASSERT_EQUAL_MESSAGE("Namespace attribute uri mismatch", std::string(uri), std::string(stag.getURI(0)));
+          CPPUNIT_ASSERT_EQUAL_MESSAGE("Local attribute name mismatch", std::string("foo"), std::string(stag.getLocalName(1)));
           break;
         case IXmlPullParser::END_TAG:
           parser->readEndTag(etag);
           output.appendf("</%s>", etag.getQName());
-          if (!streq(etag.getLocalName(), "UnitTest"))
-          {
-            fprintf(stdout, "\nTest(%s) unexpected local end tag - '%s' instead of '%s'", cfg.name, etag.getLocalName(), "UnitTest");
-            failed = true;
-          }
-          if (!streq(stag.getUri(), uri))
-          {
-            fprintf(stdout, "\nTest(%s) end tag uri mismatch - '%s' instead of '%s'", cfg.name, etag.getUri(), uri);
-            failed = true;
-          }
+          CPPUNIT_ASSERT_EQUAL_MESSAGE("Local end tag mismatch", std::string("UnitTest"), std::string(etag.getLocalName()));
+          CPPUNIT_ASSERT_EQUAL_MESSAGE("End tag uri mismatch", std::string(uri), std::string(etag.getUri()));
           break;
         }
       }
-      if (!streq(input, output))
-      {
-        fprintf(stdout, "\nTest(%s) input not equal to output:\n    in:  '%s'\n    out: '%s'", cfg.name, input.str(), output.str());
-        failed = true;
-      }
-      return !failed;
+      CPPUNIT_ASSERT_EQUAL_MESSAGE("Input not equal to output", std::string(input), std::string(output));
     }
 
     void testSkipDocument()
     {
-      const char* test = "skip-document";
       std::unique_ptr<IFragmentedXmlPullParser> parser(fxpp::createParser());
       const char* input = R"!!!(<Root><Child/></Root>)!!!";
       parser->setInput(input, int(strlen(input)));
       try
       {
         parser->skipSubTree();
-        if (parser->next() != XmlPullParser::END_DOCUMENT)
-        {
-          fprintf(stdout, "\nTest(%s) did not leave parser at expected position", test);
-          CPPUNIT_ASSERT(false);
-        }
+        CPPUNIT_ASSERT_MESSAGE("Parser not left at expected position", parser->next() == XmlPullParser::END_DOCUMENT);
       }
       catch (XmlPullParserException& xe)
       {
-        fprintf(stdout, "\nTest(%s) failed with exception: %s", test, xe.what());
-        CPPUNIT_ASSERT(false);
+        CPPUNIT_FAIL(VStringBuffer("Parser threw an exception: %s", xe.what()));
       }
```

Comment:
This may not be necessary. Given the `what` method, this may be derived from `std::exception` and the framework may already catch and report the exception. Catching `IException *` would be required because the framework doesn't know how to handle it.